src/transformers/pipelines.py

```diff
@@ -712,10 +712,23 @@ class TextClassificationPipeline(Pipeline):
             on the associated CUDA device id.
     """
 
+    def __init__(self, return_all_scores: bool = False, **kwargs):
+        super().__init__(**kwargs)
+
+        self.return_all_scores = return_all_scores
```

Comment:
should this be a kwarg on call? since init is expensive people might want to call init and then figure out the return type they need. `compute_and_score` could also be a separate method to simplify return types.

Reply:
I was wondering about putting the kwarg on the `call`, but other pipelines seem to pass the options on the constructor which seems more natural to me for this particular option.